packages/lesswrong/server/crossposting/handlers.ts

```diff
@@ -179,7 +183,25 @@ export const addV2CrosspostHandlers = (app: Application) => {
     updateCrosspostRoute,
     async function updateCrosspostHandler(context, {token}) {
       const {postId, ...postData} = await updateCrosspostToken.verify(token);
-      await context.Posts.rawUpdateOne({_id: postId}, {$set: postData});
+
+      const post = await Posts.findOne({_id: postId});
+      if (!post) {
+        throw new InvalidPostError();
+      }
+
+      const currentUser = await Users.findOne({_id: post.userId});
```

Comment:
Hmm we are just using the primary author's `userId`? I'm not a fan of creating a weird edge case, but also I'm not sure what we can do, since we don't necessarily have a matching user on this site for the editor on the other site (for example, if an admin edited the post). Seems fine then.